stl/inc/vector

```diff
@@ -1179,9 +1178,8 @@ public:
                 _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
                 _ASAN_VECTOR_CREATE;
             } else {
-                _ASAN_VECTOR_GUARD(_Newsize);
+                _ASAN_VECTOR_GUARD_CREATE;
                 _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
-                _ASAN_VECTOR_RELEASE;

```

Comment:
Check my reasoning:
When `_Clear_and_reserve_geometric` returns, ASAN believes the entire block is accessible. `_ASAN_VECTOR_GUARD_CREATE` records `0` for `_Target_size`, but doesn't change the annotation. If `_Uninitialized_fill_n` succeeds, the annotator destructor calls `_Apply_annotation(first, end, first, last)` which redundantly marks `[first, last)` accessible and fails to mark `[last, end)` inaccessible. Otherwise - if `_Uninitialized_fill_n` fails - the destructor calls `_Apply_annotation(first, end, first, first)` which does nothing. In either case, the end result is that the entire block is accessible when what we really want is for only `[first, last)` to be accessible.

Am I misunderstanding how `_s_a_c_c` works?

Reply:
No, I just wrote the code wrong. I should have been recording the capacity of the vector as the "`_Target_size`" but I was instead recording the size.

Either way I'm implementing the try/catch approach that Miscco suggested above, unless you prefer my approach.